src/mlx_omni_server/rerank/rerank_service.py

```
@@ -0,0 +1,196 @@
+        return model_id
+
+    # Check if it looks like a HuggingFace model ID (contains /)
+    if "/" in model_id and not model_id.startswith(("/", "./")):
+        try:
+            # Download/cache the model and return the local path
+            local_path = snapshot_download(
+                repo_id=model_id,
+                local_files_only=False,  # Allow downloading if not cached
+            )
+            logger.info(f"Resolved HuggingFace model {model_id} to {local_path}")
+            return local_path
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)